Some stuff needed for my project

francescortiz opened the pull request:
These are all the changes I had to do in order to be able to use elm-ethereum in my project.

The most important change to highlight because it won't trigger any compilation error is that elm Time uses miliseconds while blockchain timestamps are in seconds, so I added `(*) 1000` to the output value.

cmditch commented:
Excellent, thanks for the PR.

Good catch on the time mismatch.

Can you elaborate on reasoning behind these changes:
- Log.removed defaulting to `False`
- Block.nonce being `Maybe String`
- `addSigned0x` in Internal.Utils

Thanks!

francescortiz commented:
> Log.removed defaulting to False

I had set it to False because parity node (parity version 11.4) does not provide it for all cases. Maybe our version is a bit outdated, and since the definition states that it has to be either True or False I assumed that lack of value means False. Do you think it should be a Maybe?

> Block.nonce being Maybe String

According to the definition, `nonce` is null when it is a pending block. I encountered this.
[https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblockbyhash]

> addSigned0x in Internal.Utils

When I implemented `int : EvmDecoder BigInt` I just replicated `uint : EvmDecoder BigInt`. It was callign add0x. I addapted it to the same process but when there can be a minus sign. I see that your implementation does not use add0x and I don't use the function anywhere else, so it can be removed. Should I removed the addsigned0x?

Thank you!

cmditch commented:
Looks like Eth.getBlockByNumber supports passing strings like "pending" and "latest" etc. So I need to think about the API for this, and indeed if/how the BlockType will change.

As far as decoding Int's, unfortunately it's not as simple as replicating `uint` and adding a "-". The way signed integers are implemented in Ethereum's ABI is by using [twos-complement](https://stackoverflow.com/questions/1049722/what-is-2s-complement). Fortunately, I [just added](https://github.com/cmditch/elm-ethereum/blob/master/src/Evm/Decode.elm#L190) support for this.

I'll look more into the Log.removed parity thing. Ideally we don't have to change the library to support every node's eccentricities. But I understand the hard spot this library is in with regards to that, and it might be unescapable.

If you could update your PR to remove `addSigned0x` that'd be great. In the mean time I'll work out the API to support pending blocks, as well as the Parity Log thing.

francescortiz commented:
Done. Thank you!
